cmake/toolchain.cmake

```diff
@@ -1,4 +1,8 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+option(MBEDIDE "Use Arm compiler from Mbed Studio" OFF)
+if(MBEDIDE)
+    set(MBED_STUDIO_ARM_COMPILER "--ide=mbed")
```

Comment:
What happens if this is passed to a version of Arm compiler that isn't from Mbed Studio?

Reply:
It is a command line option which is disabled by default. It is expected to be used to compile using Arm compiler shipped with Mbed studio. If this option is used by mistake then Arm compiler will through a warning message indicating unused command line option.

Reply:
OK, so we have to avoid passing this to skip the warning option.